xml/tuning-architecture.xml

```diff
@@ -0,0 +1,338 @@
+   <dm:translation>yes</dm:translation>
+   <dm:release>SES 7</dm:release>
+  </dm:docmanager>
+ </info>
+ <para>
+   Architectural tuning includes aspects that range from the low level design
+   of the systems being employed, up to macro level decisions about network
+   topology and cooling. Not all of these can be covered in this work, but
+   guidance will be given where possible.
+ </para>
+ <para>
+   From the top-down view, it is important to think about the physical
+   location of nodes, the connectivity available between them, and
+   implications of such items as power routing, and fire compartments.
+   However, from a performance perspective, it is most important to
+   think in terms of the connectivity and what a write actually looks like
+   from the cluster perspective.
+ </para>
+ <para>
+   The desire of architectural tuning is to take control of where the
+   performance bottlenecks lie. In most cases, it is preferable for the
+   bottleneck to lie with the storage device, as that creates a predictable
+   pattern of performance degradation and associated limitations. Placing
+   the bottleneck in other areas, such as CPU or network, may create
+   inconsistent behavior when resources are stressed. This is due to the
+   possibility of other processes running, whether recovery, rebalancing,
+   garbage collection, and causing inconsistent behavior due to being
+   network or CPU bound. When storage is the bottleneck, the degradation
+   results in longer response times to queued requests, making this the most
+   desireable point at which to place the bottleneck.
+ </para>
+ <sect1 xml:id="tuning-network">
+   <title>Network</title>
+   <para>
+     The network is the backbone of the cluster, and a failure to implement
+     it in a robust manner can cripple the performance of the cluster, no
+     matter what other steps are taken. From a best practices perspective,
+     this entails implementing a fault-tolerant switching infrastructure that
+     can scale the aggregate core bandwidth as the cluster grows. For very
+     large clusters, this may entail a leaf and spine architecture, while for
+     smaller clusters, it may look like a more familiar hub-and-spoke or
+     mesh network.
+   </para>
+   <para>
+     No matter which network architecture is chosen, it is important to
+     carefully examine each hop along the path and consider the maximum
+     network traffic that could occur during adverse conditions.
+   </para>
+   <para>
+     An example of a bad decision here would be to have a multi-rack cluster
+     where each rack contains 16 storage nodes of 24x 7.2k drives where each
+     node is connected to a pair of stacked Top-of-Rack (TOR) switches via
+     2x25Gb connections. This connectivity is sufficient for 3GB/s per
+     connection or 6GB/s total for the node, which is near the max that a
+     7.2k drive will sustain. The bad decision comes in ONLY using 4x of
```

Comment:
Is "7.2k" drive specific enough? Do we need to use "7200 rpm" or something clearer?

Also, I suggest replacing the caps on ONLY with [em]only[/em] or similar.

Reply:
I've pinged David to see what he thinks

Reply:
I am fine with both suggestions.